src/freenas/usr/local/bin/truenas-grub.py

```diff
@@ -19,18 +19,22 @@ def get_serial_ports():
     # 22.02-RC.2 use real hostid. We need to be able to boot both of these configurations.
     config = [
         'GRUB_DISTRIBUTOR="TrueNAS Scale"',
+        'GRUB_TIMEOUT=10',
         'GRUB_CMDLINE_LINUX_DEFAULT="libata.allow_tpm=1 amd_iommu=on iommu=pt '
         'kvm_amd.npt=1 kvm_amd.avic=1 intel_iommu=on zfsforce=1'
         f'{f" {kernel_extra_options}" if kernel_extra_options else ""}"',
     ]
 
-    terminal = ["console"]
+    terminal_output = ["gfxterm"]
```

Comment:
Do we need to use "gfxterm" if we are not using "serial"?  Does it give any benefits?  I would guess "console" to be more universal.  I guess it may react on serial console redirection setting in BIOS, at least I would check it.

Reply:
Using 'gfxterm' instead of 'console' helps in two different problems. 'gfxterm' only outputs to graphics console and it would fail if there is not graphics option. Using 'gfxterm' and serial in this way would help in handling both EFI and legacy BIOS systems independent of serial redirection setting in BIOS.

Reply:
> 'gfxterm' only outputs to graphics console and it would fail if there is not graphics option.

Right. Is it good in some way to fail there and get no console?

> Using 'gfxterm' and serial in this way would help in handling both EFI and legacy BIOS systems independent of serial redirection setting in BIOS.

Again I am not sure why cooperation with BIOS settings is bad?  If I have a system with console redirection in BIOS, I would prefer to have the redirection in GRUB menu also, sure without double output.  Speaking about EFI and BIOS it would be good to have the logic in comments for both cases if they behave differently.

Reply:
> Right. Is it good in some way to fail there and get no console?

We should not fail and have no console in any case. This was the reason to patch the GRUB config generation. With serial console enabled, if 'gfxterm' fails, we fallback to serial console only. Without serial console, if 'gfxterm' fails we do set terminal_output to console in that case. And this happens only if serial is disabled.

> Again I am not sure why cooperation with BIOS settings is bad?

We basically cannot get rid of doubled output while using console and working with serial console options. We cannot know if serial redirection is enabled in BIOS. Moreover, this makes the serial console selection from WebUI inconsistent. Console writes to all serial ports in EFI while user could have selected a specific port from WebUI and complain that all serial ports have console.